ui/apps/everest/src/hooks/api/namespaces/useNamespaces.ts

```diff
@@ -49,11 +51,102 @@ export const useDBEnginesForNamespaces = (retrieveUpgradingEngines = false) => {
   const queryResults = useQueries({
     queries,
   });
+
+  const refetchAll = useCallback(() => {
+    queryResults.forEach((result) => result.refetch());
+  }, [queryResults]);
+
   const results = queryResults.map((item, i) => ({
     namespace: namespaces[i],
     ...item,
   }));
-  return results;
+  return { results, refetchAll };
+};
+
+export interface DbEngineForNamedpaceExpanded {
+  dbEngine?: DbEngine;
+  namespace: string;
+}
+export interface DbEnginesForDbTypeExpanded {
+  type: DbEngineType;
+  available: boolean;
+  dbEngines: DbEngineForNamedpaceExpanded[];
+}
+export const useDBEnginesForDbEngineTypes = (
+  dbEngineType?: DbEngineType
+): [
+  dbEnginesFoDbEngineTypes: DbEnginesForDbTypeExpanded[],
+  dbEnginesFoDbEngineTypesFetching: boolean,
+  refetch: () => void,
+] => {
+  const { results: dbEnginesForNamespaces, refetchAll } =
+    useDBEnginesForNamespaces();
+  const dbEnginesFetching = dbEnginesForNamespaces.some(
+    (result) => result.isFetching
+  );
+
+  const dbEngineTypes = useMemo(
+    () =>
+      dbEngineType
+        ? [dbEngineType]
+        : (Object.keys(DbEngineType) as Array<keyof typeof DbEngineType>).map(
+            (type) => DbEngineType[type]
+          ),
+    [DbEngineType]
+  );
+
+  const availableDbEngineTypes = useMemo(() => {
+    if (!dbEnginesFetching) {
+      return dbEngineTypes.map((type) => {
+        const dbEnginesForDbType = dbEnginesForNamespaces.reduce(
```

Comment:
we have an array of responces for each namespace, and we need to map it to array of dbTypes, where for each we have info about all namespaces, where this kind of db is present (and dbEngines data), so in this case reduce looks like solution, but if we want to improve the readability of the code, of course we can replace this part with map/filter, etc.